test/test_io.py

```diff
@@ -43,6 +72,27 @@ def test_create_destroy(self):
         del bootstrap
         self.assertTrue(bootstrap_shutdown_event.wait(TIMEOUT))
 
+    def test_create_destroy_singleton(self):
+        bootstrap = ClientBootstrap.get_or_create_static_default()
+
+        # ensure shutdown_event fires
+        bootstrap_shutdown_event = bootstrap.shutdown_event
+        del bootstrap
+
+        ClientBootstrap.release_static_default()
+        EventLoopGroup.release_static_default()
+        DefaultHostResolver.release_static_default()
+
+        self.assertTrue(bootstrap_shutdown_event.wait(TIMEOUT))
+
+    def test_init_singleton_is_singleton(self):
+        client_one = ClientBootstrap.get_or_create_static_default()
+        client_two = ClientBootstrap.get_or_create_static_default()
+        self.assertTrue(client_one == client_two)
+        ClientBootstrap.release_static_default()
+        EventLoopGroup.release_static_default()
+        DefaultHostResolver.release_static_default()
+
 
 class ClientTlsContextTest(NativeResourceTest):
     def test_init_defaults(self):
```

Comment:
add support for static-defaults everywhere EventLoopGroup, HostResolverBase, and ClientBootstrap are taken:
- [http.HttpClientConnection.new()](https://github.com/awslabs/aws-crt-python/blob/cc651452fa1900c140fedde0be65962b5b897396/awscrt/http.py#L85)
    -  ugh, looks like once upon a time  **someone** already tried adding some hideous version of static defaults that would create NEW objects if the bootstrap wasn't specified. That is horrible. That is a bloated waste of resource. Oh, [that was me](https://github.com/awslabs/aws-crt-python/blame/cc651452fa1900c140fedde0be65962b5b897396/awscrt/http.py#L129) 😐
- [mqtt.Client()](https://github.com/awslabs/aws-crt-python/blob/cc651452fa1900c140fedde0be65962b5b897396/awscrt/mqtt.py#L141)
- [s3.S3Client()](https://github.com/awslabs/aws-crt-python/blob/cc651452fa1900c140fedde0be65962b5b897396/awscrt/s3.py#L90)
- [eventstream.rpc.ClientConnection.connect()](https://github.com/awslabs/aws-crt-python/blob/cc651452fa1900c140fedde0be65962b5b897396/awscrt/eventstream/rpc.py#L261)